File: Java/src/main/java/com/gildedrose/Sulfuras.java
package com.gildedrose;

public class Sulfuras implements Goods {

	@Override
	public void updateQuality(Item item) {
		// Do nothing because Sulfuras being a legendary item, never has to be sold or
		// decreases in quality.

	}

	@Override
	public void updateSellInDays(Item item) {
		// Do nothing because Sulfuras being a legendary item, never has to be sold or
		// decreases in quality.
	}

	@Override
	public void updateQualityForExpiredItem(Item item) {
		// Do nothing because Sulfuras being a legendary item, never has to be sold or
		// decreases in quality.		
	}

}

File: csharpcore/GildedRose/Models/Sulfuras.cs
using GildedRose.Abstraction;
using GildedRoseKata;

namespace GildedRose.Models
{
    public class Sulfuras : Item, ICustomMethod
    {
        public int SellDaysGone { get; set; }

        public void UpdateQuality()
        {
            // do thing
        }

        public void UpdateSellIn()
        {
            // do nothing
        }
    }
}
